### PR TITLE
mz: update README.md + MAINTAINERS.md + CHANGELOG.md

### PR DESCRIPTION
**Pull request by joacoc**
### Motivation

This PR adds missing documents for `mz`:
- `README.md`
- `CHANGELOG.md`

It also includes a subtle fix to `MAINTAINERS.md`.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/mz/MAINTAINERS.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ cargo test -p mz
 
 3. Run `cargo run -p mz` once to update `Cargo.lcok`
 
-3. Open a PR with your change and get it merged:
+4. Open a PR with your change and get it merged:
```
Comment (morsapaes) [inline src/mz/MAINTAINERS.md:20]: Looks like you missed the one below (5.)!

**Review by morsapaes — APPROVED** [review]

File: src/mz/README.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+# The default region to use for the Acme Corp profile.
+region = "aws/us-east-1"
+
+# Profile for Hooli.
+[profiles.hooli]
+app-password = "mzp_a48df1039ecb2d94c..."
+region = "aws/eu-west-1"
+```
+
+## Dependencies
```
Comment (morsapaes) [inline src/mz/README.md:166]: These dependencies are for local development...right? It's confusing to list them here.

**Review by morsapaes — CHANGES_REQUESTED** [review]

File: src/mz/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Change Log
+
+All notable changes to the `mz` CLI will be documented in this file.
+
+## [0.1.3] - 2023-09-08
+
+This new version contains a big revamp of the `mz` CLI. It includes updated code with an entirely different pattern for `profiles`.
```
[the remark this answers is not in the document]
Comment (joacoc) [reply src/mz/CHANGELOG.md:7]: Is it a refactor if it implies external effects?

File: src/mz/MAINTAINERS.md [hunk 4]
```diff
@@ -1,13 +1,23 @@
...
 
+## Testing the CLI
+
+All the tests for the CLI reside in a single function inside `./tests/e2e.rs`. To test locally run:
+
+```
+cargo test -p mz
+```
+
 ## Cutting a new release
```
[the remark this answers is not in the document]
Comment (joacoc) [reply src/mz/MAINTAINERS.md:11]: I was planning to set the `1.0` before making it public to users (after the private test internally.) Does it makes sense?

File: src/mz/README.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+   ```
+   psql (14.2)
+   Type "help" for help.
+
+   you@corp.com=#
+   ```
+
+   Substitute `aws/us-east-1` with the name of an enabled region in your
+   organization. If you don't yet have an enabled region, use
+   [`mz region enable`](reference/region) to enable one.
```
[the remark this answers is not in the document]
Comment (joacoc) [reply src/mz/README.md:79]: Mhh. Interesting. This error is new. You should be able to enable a region that is already enabled. In anycase, it doesn't have any positive impact for the customer and can take down `environmentd` for a few minutes. So the best thing will be to just say it is not possible to do, and only enable it for internal  devs with a flag.

File: src/mz/README.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+## Getting started
+
+1. Install `mz`:
+
+   ```shell
+   # On macOS:
+   brew install materializeinc/materialize/mz
+   # On Ubuntu/Debian:
+   curl -fsSL https://dev.materialize.com/apt/materialize.sources | sudo tee /etc/apt/sources.list.d/materialize.sources
```
[the remark this answers is not in the document]
Comment (joacoc) [reply src/mz/README.md:35]: Issue created: https://github.com/MaterializeInc/materialize/issues/21321

File: src/mz/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (joacoc) [reply src/mz/README.md:79]: Issue created: https://github.com/MaterializeInc/materialize/issues/21322

File: src/mz/README.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+   sudo apt update
+   sudo apt install materialize-cli
+   ```
+
+2. Log in to your Materialize account:
+
+   ```shell
+   mz profile init
```
[the remark this answers is not in the document]
Comment (joacoc) [reply src/mz/README.md:43]: Issue created: https://github.com/MaterializeInc/materialize/issues/21326

File: src/mz/README.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+SUBCOMMANDS:
+    app-password    Manage app passwords for your user account
+    config          Manage global configuration parameters for the CLI
+    help            Print this message or the help of the given subcommand(s)
+    profile         Manage authentication profiles for the CLI
+    region          Manage regions in your organization
+    secret          Manage secrets in a region
+    sql             Execute SQL statements in a region
+    user            Manage users in your organization
```
[the remark this answers is not in the document]
Comment (joacoc) [reply src/mz/README.md:24]: Issue created: https://github.com/MaterializeInc/materialize/issues/21327

File: src/mz/CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (morsapaes) [reply src/mz/CHANGELOG.md:7]: That's a good point, I commented before noticing that this also introduces new functionality. Edited the suggestion!

**Review by morsapaes — APPROVED** [review]

File: src/mz/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (joacoc) [reply src/mz/README.md:35]: @morsapaes can I ask you which Linux dist are you using?

**Pull request merged**